libraries/voicerecorder/api/src/main/kotlin/io/element/android/libraries/voicerecorder/api/VoiceRecorderState.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.voicerecorder.api
+
+import java.io.File
+
+sealed class VoiceRecorderState {
+    /**
+     * The recorder is idle and not recording.
+     */
+    data object Idle : VoiceRecorderState()
+
+    /**
+     * The recorder is currently recording.
+     *
+     * @property level The current audio level of the recording as a fraction of 1.
+     */
+    data class Recording(val level: Double) : VoiceRecorderState()
```

Comment:
I couldn't find a source for this, but I think operations involving float and double have similar performance nowadays, since 64 bit processors became the norm? But happy to be corrected!